test/framework/src/main/java/org/elasticsearch/transport/DisruptableMockTransport.java

```diff
@@ -233,11 +233,11 @@ public String toString() {
         });
     }
 
-    protected void onDisconnectedDuringSend(long requestId, String action, DisruptableMockTransport destinationTransport) {
-        destinationTransport.execute(getDisconnectException(requestId, action, destinationTransport.getLocalNode()));
+    private void onDisconnectedDuringSend(long requestId, String action, DisruptableMockTransport destinationTransport) {
+        execute(getDisconnectException(requestId, action, destinationTransport.getLocalNode()));
```

Comment:
This line is the fix, everything else is just noise. Looks simple, but wow this took hours to pin down.